### PR TITLE
Length scalars

### PR DESCRIPTION
**Pull request by kovrus**

### CODE REVIEW

File: docs/sql/scalar.txt [hunk 1]
```diff
@@ -101,6 +101,93 @@ which defines the length of the substring::
...
+
+octet_length('string') returns integer
+--------------------------------------
+
+The ``octet_length`` function returns the number of bytes (octets) in the input
+string.
+
+Example::
+
+    cr> select octet_length('©rate.io') from sys.cluster;
```
Comment (chaudum) [inline docs/sql/scalar.txt]: I would swap the examples like you did with `bit_length`

File: CHANGES.txt [hunk 2]
```diff
@@ -5,6 +5,9 @@ Changes for Crate
 Unreleased
 ==========
 
+ - Add ``char_length``, ``bit_length`` and ``octet_length`` scalar
```
Comment (chaudum) [inline CHANGES.txt]: Added ...

File: sql/src/main/java/io/crate/operation/scalar/LengthFunction.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    }
+
+    protected static FunctionInfo createInfo(String functionName, DataType dataType) {
+        return new FunctionInfo(
+                new FunctionIdent(functionName, ImmutableList.of(dataType)), DataTypes.INTEGER
+        );
+    }
+
+    protected BytesRef evaluateInput(Input[] args) {
+        Preconditions.checkArgument(args.length == 1);
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/operation/scalar/LengthFunction.java]: could be changed to an assertion as this is basically ensured to be the case due to the function registration

File: sql/src/main/java/io/crate/operation/scalar/LengthFunction.java [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+        Object string = args[0].value();
+        if (string == null) {
+            return null;
+        }
+        return (BytesRef) string;
+    }
+
+    @Override
+    public Symbol normalizeSymbol(Function symbol) {
+        Preconditions.checkArgument(symbol.arguments().size() == 1);
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/operation/scalar/LengthFunction.java]: same here

**Comment by mfussenegger** [comment]
ack

**Pull request merged**